src/module/token/minting/ClaimableERC721.sol

```diff
@@ -188,22 +183,24 @@ contract ClaimableERC721 is Module, EIP712, BeforeMintCallbackERC721, IInstallat
     {
         ClaimParamsERC721 memory _params = abi.decode(_data, (ClaimParamsERC721));
 
-        address currency;
-        uint256 pricePerUnit;
+        _validateClaimCondition(_to, _quantity, _params.currency, _params.pricePerUnit, _params.recipientAllowlistProof);
 
-        if (_params.signature.length == 0) {
-            _validateClaimCondition(
-                _to, _quantity, _params.currency, _params.pricePerUnit, _params.recipientAllowlistProof
-            );
-            currency = _params.currency;
-            pricePerUnit = _params.pricePerUnit;
-        } else {
-            _validateClaimRequest(_to, _quantity, _params.request, _params.signature);
-            currency = _params.request.currency;
-            pricePerUnit = _params.request.pricePerUnit;
-        }
+        _distributeMintPrice(msg.sender, _params.currency, _quantity * _params.pricePerUnit);
+    }
+
+    /// @notice Callback function for the ERC721Core.mint function.
+    function beforeMintWithSignatureERC721(address _to, uint256 _startTokenId, uint256 _quantity, bytes memory _data)
```

Comment:
@GWSzeto I also need a `encodeBytesBeforeMintWithSignature` in the Abu so I can generate the encoding properly. Just like regular mint

Reply:
addressed